### PR TITLE
Nova Compute Ironic failover procedure

### PR DESCRIPTION
**Pull request by dougszumski**
- Moving from multiple instances to a single instance
- How to re-deploy the service

### CODE REVIEW

File: doc/source/operations/nova-compute-ironic.rst [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+
+Note that an ``-ironic`` postfix is added to the hostname. This comes from
+a convention in Kolla Ansible. It is worth making this change ahead of time,
+even if you don't need to immediately re-deploy the service.
+
+It is also possible to use an arbitrary ``host`` name, but you will need
+to edit the database again. That is an optional exercise left for the reader.
+See [1] for further details.
+
+TODO: Investigate KA bug with assumption about host field.
```
Comment (dougszumski) [inline doc/source/operations/nova-compute-ironic.rst]: Need to do something about this before merging, as it breaks 'waiting for compute hosts to register'.

File: doc/source/operations/nova-compute-ironic.rst [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+8. Ensure that all Ironic nodes are using the single remaining Nova Compute
+   Ironic instance. Eg. Baremetal nodes in use by compute instances will not
+   fail over to the remaining Nova Compute Ironic service. Here, the active
+   service is running on ``controller1``:
+
+  .. code-block:: console
+
+    $ ssh controller1
+    $ sudo docker exec -it mariadb mysql -u nova -p$(sudo grep 'mysql+pymysql://nova:' /etc/kolla/nova-api/nova.conf | awk -F'[:,@]' '{print $3}')
```
Comment (dougszumski) [inline doc/source/operations/nova-compute-ironic.rst:144]: We could do this with the new nova manage CLI command in Caracal, but at least this works for everyone. When I have an Ironic env, I can update with instructions.

File: doc/source/operations/nova-compute-ironic.rst [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+Typically you will see either three or one. By default the host will
+marked with a postfix, eg. ``controller1-ironic``. If you find more than
+one, you will need to remove some instances. You must complete the
+following section.
+
+Moving from multiple Nova Compute Instances to a single instance
+----------------------------------------------------------------
+
+1. Decide where the single instance should run. Typically, this will be
```
Comment (Alex-Welsh) [inline doc/source/operations/nova-compute-ironic.rst]: What would be a good method of deciding where the single instance should run?

File: doc/source/operations/nova-compute-ironic.rst [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+the new host.
+
+To work around this you should set the ``host`` field in ``nova.conf`` to a
+constant, such that the new Nova Compute Ironic instance comes up with the
+same name as the one it replaces.
+
+For example, if the original instance resides on ``controller1``, then set the
+following in ``etc/kayobe/nova.yml``:
+
+.. code-block:: console
```
Comment (Alex-Welsh) [inline doc/source/operations/nova-compute-ironic.rst]:
```suggestion
.. code-block:: yaml
```

File: doc/source/operations/nova-compute-ironic.rst [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+  kolla_nova_compute_ironic_static_host_name: controller1-ironic
+
+Note that an ``-ironic`` postfix is added to the hostname. This comes from
```
Comment (dougszumski) [inline doc/source/operations/nova-compute-ironic.rst:205]: I'm adding some notes on running the Kayobe commands to re-deploy the service. And considerations that need to be made if the old host is offline etc. (breaking the service deploy, pending 'Kayobe reliability' changes, etc.

File: doc/source/operations/nova-compute-ironic.rst (hunk 3 above)
Comment (dougszumski) [reply doc/source/operations/nova-compute-ironic.rst]: I've added some words - generally any controller.

**Comment by dougszumski** [comment]
Needs updating, pending this: https://review.opendev.org/c/openstack/kolla-ansible/+/912483

**Review by markgoddard — COMMENTED** [review]
A good write up. Should we merge and update once the kolla-ansible patch lands and is backported?

**Comment by dougszumski** [comment]
> A good write up. Should we merge and update once the kolla-ansible patch lands and is backported?

Thanks for taking the time. That sounds good, i'll make sure it's updated quickly.

**Review by markgoddard — APPROVED** [review]

**Pull request merged**